### PR TITLE
Suntoh/feat12/gamepage

### LINKED ISSUE #12: [FE] Game Room
Player’s name and score are appeared on the game client.

### PR DESCRIPTION
**Pull request by Suntoh**
## What I did

**create game page**
- create piano interface which can be pressed
- map keybind to the piano key
- display pressed note as bubble on screen
- send player from lobby to gameroom
- send pressed note, time to other player in the same room
- feature: resign button, back to lobby page
- feature: sound played when piano key pressed
- feature: pressed 2 key at a time

## Note

need to create the play along page(can be in the same page or create a separate)
recieve the pressed note and check score there

### CODE REVIEW

File: apps/client/src/app/lobby/[roomId]/page.tsx [hunk 1]
```diff
@@ -58,6 +58,7 @@ export default function LobbyPage() {
   const startGame = () => {
     if (players.length >= 2) {
       socket.emit('start_game', roomId);
+      redirect(`/game/${roomId}`);
```
Comment (Imeany12) [inline apps/client/src/app/lobby/[roomId]/page.tsx]: I don't got redirect here but, received log from server.
Is it suppose to behave like this?

File: apps/client/src/app/set/page.tsx
Comment (Imeany12) [inline apps/client/src/app/set/page.tsx:1]: I can't set the keybind TT

https://github.com/user-attachments/assets/6358d2cc-4eb8-4f6e-8357-122b670791a1

**Review by Imeany12 — COMMENTED** [review]
Just a head up if you want to handle triggering reload when new user join or leave
Screenshot 2567-10-21 at 23 19 17

File: apps/client/src/app/lobby/[roomId]/page.tsx (hunk 1 above)
Comment (Suntoh) [reply apps/client/src/app/lobby/[roomId]/page.tsx]: changed to router.push   Works better

File: apps/client/src/app/set/page.tsx
Comment (Suntoh) [reply apps/client/src/app/set/page.tsx:1]: At first I console.log to check that the function works so I am waiting to use it with the server.
I implement the localStorage for now.
In final stage I would like to send it to server with the usere account

**Review by Imeany12 — APPROVED** [review]
LGTM 🚀

**Pull request merged**